src/Microsoft.TestPlatform.ObjectModel/TestProperty/CustomKeyValueConverter.cs

```diff
@@ -38,11 +46,12 @@ public override object ConvertFrom(ITypeDescriptorContext context, CultureInfo c
         // KeyValuePairs are used for traits. 
         if (value is string data)
         {
+            // PERF: The values returned here can possibly be cached, but the benefits are very small speed wise,
+            // and it is unclear how many distinct objects we get, and how much memory this would consume. I was seeing around 100ms improvement on 10k tests.
+
             using var stream = new MemoryStream(Encoding.Unicode.GetBytes(data));
             // Converting Json data to array of KeyValuePairs with duplicate keys.
-            var serializer = new System.Runtime.Serialization.Json.DataContractJsonSerializer(typeof(TraitObject[]));
-            var listOfTraitObjects = serializer.ReadObject(stream) as TraitObject[];
-
+            var listOfTraitObjects = _serializer.ReadObject(stream) as TraitObject[];
             return listOfTraitObjects.Select(i => new KeyValuePair<string, string>(i.Key, i.Value)).ToArray();
```

Comment:
Potential NRE, you are doing a `as` cast.

Reply:
Fixed